.github/workflows/ci.yaml

```diff
@@ -0,0 +1,147 @@
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+name: Gridiron
+
+jobs:
+  check:
+    name: Check
+    runs-on: ubuntu-18.04
+    strategy:
+      matrix:
+        rust:
+          - stable
+          - beta
+          - 1.38.0 #MSRV
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: ${{ matrix.rust }}
+          override: true
+      - uses: actions-rs/cargo@v1
+        with:
+          command: check
+
+  build_linux:
+    name: (Linux) Platform Builds 
+    runs-on: ubuntu-18.04
+    strategy:
+      matrix:
+        target:
+          - armv7-linux-androideabi     # Android ARM7
+          - aarch64-linux-android       # Android x64
+          - x86_64-unknown-linux-musl   # Alpine Linux x86_64
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          target: ${{ matrix.target }}
+          override: true
+      - uses: actions-rs/cargo@v1
+        with:
+          use-cross: true
+          command: build
+          args: --release --target=${{ matrix.target }}
+
+  build_macos:
+    name: (MacOS X) Platform Builds 
+    runs-on: macos-10.15
+    strategy:
+      matrix:
+        target:
+          - armv7-apple-ios
+          - aarch64-apple-ios
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          target: ${{ matrix.target }}
+          override: true
+      - uses: actions-rs/cargo@v1
+        with:
+          use-cross: true
+          command: build
+          args: --release --target=${{ matrix.target }}
+
+  build_win:
+    name: (Windows) Platform Builds 
+    runs-on: windows-2019
+    strategy:
+      matrix:
+        target:
+          - x86_64-pc-windows-msvc
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          target: ${{ matrix.target }}
+          override: true
+      - uses: actions-rs/cargo@v1
+        with:
+          command: build
+          args: --release --target=${{ matrix.target }}
+
+  test:
+    name: Test Suite
+    runs-on: ubuntu-18.04
+    strategy:
+      matrix:
+        rust:
+          - stable
+          - 1.38.0
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: ${{ matrix.rust }}
+          override: true
+      - uses: actions-rs/cargo@v1
+        with:
+          command: test
+
+  # bench_build:
```

Comment:
Does this need a comment about why it's disabled?

Reply:
I mentioned it in the PR, but also added a comment in the code. The benches are not currently building and I don't want to address that in this PR.